waitpid/test_waitpid.c

```diff
@@ -42,22 +42,23 @@ TEST(test_waitpid, waitpid_wnohang)
 	int res, i, j;
 	char *const arg[][2] = { { "exec_while_process", NULL }, { "exec_sum_process", NULL } };
 
-	/* 1 iteration for easier problem reproduction, in the future there will be more irerations */
+	/* 1 iteration for easier problem reproduction, in the future there will be more iterations */
 	for (i = 1; i > 0; i--) {
 		for (j = 0; j < 2; j++) {
-			if ((pid[j] = vfork()) < 0) {
+			pid[j] = vfork();
+			if (pid[j] < 0) {
 				fprintf(stderr, "vfork function failed: %s\n", strerror(errno));
 				return;
 			}
 			/* child process j */
-			else if (!pid[j]) {
+			else if (pid[j] == 0) {
 				execv("/bin/test-waitpid", arg[j]);
 				fprintf(stderr, "exec function failed: %s\n", strerror(errno));
 				_exit(EXIT_FAILURE);
 			}
 		}
 		/* wait for process 1 to become a zombie process */
-		usleep(200000);
+		usleep(100000 * 10);
```

Comment:
That a peculiar way of writing 1 second :)
How about more common:
```suggestion
		usleep(1000 * 1000);
```

Reply:
The intent wasn't for one second, it was to have a multiple of time used in usleep in main in case of  `exec_sum_process`.